detekt/ruleset.yml

```diff
@@ -0,0 +1,10 @@
+build:
+  maxIssues: 60 # break the build if more than ten weighted issues are found
+style:
+  MagicNumber:
+    ignoreEnums: true
+complexity:
+  TooManyFunctions:
+    thresholdInClasses: 20
+    thresholdInObjects: 20
+    thresholdInFiles: 20
```

Comment:
These were increased from the defaults to allow this project to pass.

When we update to support Platform 0.22 Proofs, there will be opportunity to refactor to simplify the `DapiClient` class which currently does all the heavy lifting of parsing the data returned from all of the `get` DAPI calls.  

For example, `DapiClient.getDocuments` will take the results of a getDocuments DAPI call and then verify and convert the data to be returned.  Perhaps the verification and conversion should be done by the `GetDocumentsMethod`.  In the bigger picture, these `Method` classes can do the same (verify and covert data).  `DapiClient` shouldn't have to know anything about how data should be handled.  It's main purpose is to find a masternode and then make the DAPI call and retry if necessary.